src/to-have-display-value.js

```diff
@@ -0,0 +1,44 @@
+import {matcherHint} from 'jest-matcher-utils'
+
+import {checkHtmlElement, getMessage} from './utils'
+
+export function toHaveDisplayValue(htmlElement, expectedValue) {
+  checkHtmlElement(htmlElement, toHaveDisplayValue, this)
+  const tagName = htmlElement.tagName.toLowerCase()
+
+  if (!['select', 'input', 'textarea'].includes(tagName)) {
+    throw new Error(
+      '.toHaveDisplayValue() currently supports only input, textarea or select elements, try with another matcher instead.',
+    )
+  }
+
+  if (tagName === 'input' && ['radio', 'checkbox'].includes(htmlElement.type)) {
+    throw new Error(
+      `.toHaveDisplayValue() currently does not support input[type="${htmlElement.type}"], try with another matcher instead.`,
+    )
+  }
+
+  const value =
+    tagName === 'select'
+      ? Array.from(htmlElement)
+          .filter(option => option.selected)
+          .map(option => option.textContent)
+          .toString()
+      : htmlElement.value
+
+  return {
+    pass: value === expectedValue.toString(),
```

Comment:
Hmmm, I'm not sure if comparing as string is the best approach. I now realize that this means that for `<select multiple />` you have to pass the array with the option display texts in the same order as they appear in the list.

Reply:
Yep, its for the `multiple` case, and at the moment I don't dislike too much the order enforcing thing, is indeed the same order that the user see when using those select 🤔

Reply:
Ok, let's leave it as is and we can improve upon it later if it proves to be a problem.